src/big/json.cr

```diff
@@ -2,20 +2,6 @@ require "json"
 require "big"
 
 class JSON::Builder
-  # Writes a big integer.
-  def number(number : BigInt)
-    scalar do
-      @io << number
-    end
-  end
-
-  # Writes a big float.
-  def number(number : BigFloat)
-    scalar do
-      @io << number
-    end
-  end
```

Comment:
ftr, this erroneously overshadowed `JSON::Builder#number(number : Float)` which has a custom logic for raising on illegal float (like `NaN` and `Infinity`).

Reply:
But there's no NaN or Infinity in BigFloat... 🤔

Reply:
Hmm, atm that seems true, I'm not sure though if that's intended...

```cr
require "big"

Float64::INFINITY.to_big_f # => Program received and didn't handle signal FPE (8)
Float64::NAN.to_big_f      # => Program received and didn't handle signal FPE (8)
```

Reply:
Okay, that's a different bug. I just don't think you should remove this chunk of code.

Reply:
Well the generic implementation isn't *wrong* either. BigFloat answers `nil?` and `infinite?` with `false`, so the special branches won't match.

Reply:
Sounds good